stdlib/email/src/main/java/org/ballerinalang/stdlib/email/util/PopUtil.java

```diff
@@ -0,0 +1,213 @@
+            properties.put(ImapConstants.PROPS_IMAP_SSL_ENABLE, popConfig.getBooleanValue(PopConstants.PROPS_SSL));
+            properties.put(EmailConstants.MAIL_STORE_PROTOCOL, ImapConstants.IMAP_PROTOCOL);
+        }
+
+        properties.put(PopConstants.PROPS_USERNAME, popConfig.getStringValue(PopConstants.PROPS_USERNAME));
+        properties.put(PopConstants.PROPS_PASSWORD, popConfig.getStringValue(PopConstants.PROPS_PASSWORD));
+
+        if (log.isDebugEnabled()) {
+            log.debug("[PopUtil][getProperties] Property object created. Returning Object to PopClient");
+        }
+        return properties;
+    }
+
+    /**
+     * Generates MapValue object using the passed message.
+     *
+     * @param message received
+     * @return MapValue object
+     * @throws MessagingException If an error occurs related to messaging
+     * @throws IOException If an error occurs related to I/O
+     */
+    public static MapValue getMapValue(Message message) throws MessagingException, IOException {
+        if (log.isDebugEnabled()) {
+            log.debug("[PopUtil][getMapValue] Extracting MapValue values from Config");
+        }
+        Map<String, Object> valueMap = new HashMap<>();
+        Address[] toAddress = message.getRecipients(Message.RecipientType.TO);
+        Address[] ccAddress = message.getRecipients(Message.RecipientType.CC);
+        Address[] bccAddress = message.getRecipients(Message.RecipientType.BCC);
+        Address[] replyToAddress = message.getReplyTo();
+        ArrayValue toAddressArrayValue = new ArrayValueImpl(new BArrayType(BTypes.typeString));
+        ArrayValue ccAddressArrayValue = new ArrayValueImpl(new BArrayType(BTypes.typeString));
+        ArrayValue bccAddressArrayValue = new ArrayValueImpl(new BArrayType(BTypes.typeString));
+        ArrayValue replyToAddressArrayValue = new ArrayValueImpl(new BArrayType(BTypes.typeString));
+        if (toAddress != null) {
+            for (Address address: toAddress) {
+                toAddressArrayValue.append(address.toString());
+            }
+        }
+        if (ccAddress != null) {
+            for (Address address: ccAddress) {
+                ccAddressArrayValue.append(address.toString());
+            }
+        }
+        if (bccAddress != null) {
+            for (Address address: bccAddress) {
+                bccAddressArrayValue.append(address.toString());
+            }
+        }
+        if (replyToAddress != null) {
+            for (Address address: replyToAddress) {
+                replyToAddressArrayValue.append(address.toString());
+            }
+        }
+        String subject = getStringNullChecked(message.getSubject());
+        String messageBody = "";
+        if (message.isMimeType("text/plain")) {
+            if (message.getContent() != null) {
+                messageBody = message.getContent().toString();
+            } else {
+                messageBody = "";
```

Comment:
Shall we remove this else block? because already messageBody is an empty string;

Reply:
Fixed.